testlapack/dlaqp2.go

```diff
@@ -0,0 +1,132 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"sort"
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+type Dlaqp2er interface {
+	Dlaqp2(m, n, offset int, a []float64, lda int, jpvt []int, tau, vn1, vn2, work []float64)
+}
+
+func Dlaqp2Test(t *testing.T, impl Dlaqp2er) {
+	rnd := rand.New(rand.NewSource(1))
+	for ti, test := range []struct {
+		m, n, offset int
+	}{
+		{m: 4, n: 3, offset: 0},
+		// {m: 4, n: 3, offset: 2},
+		// {m: 4, n: 3, offset: 4},
+		{m: 10, n: 10, offset: 0},
+		// {m: 10, n: 10, offset: 5},
+		// {m: 10, n: 10, offset: 10},
```

Comment:
I found the work slice issue with a `m < n` matrix, could you please add such a test case?